demography.md

```diff
@@ -71,40 +93,61 @@ To fix this, let's add some migration events to the specific demographic history
 ## Migrations
 
 With msprime, you can specify continual rates of migrations between
-populations, as well as one-off mass migrations.
+populations, as well as admixture events, divergences and one-off mass migrations.
 
 ### Constant migration
 
 ![constant-migration-diagram](_static/tute-population-structure-1.png)
 
-Migration rates between the populations can be specified as the elements of an
-*N* by *N* numpy array, and given to {func}`msprime.sim_ancestry` via the
-``migration_matrix`` argument. The diagonal elements of this array must each be
-0, and the *(i, j)* th element specifies the expected number of migrants moving
-from population *j* to population *i* per generation, divided by the size of
-population *i*.  When this rate is small (close to 0), it is approximately
-equal to the fraction of population *i* that consists of new migrants from
-population *j* in each generation.
+Migration rates between the populations can be thought as the elements of an
+*N* by *N* numpy array, and are passed to our {class}`msprime.Demography` object individually via the {meth}`msprime.Demography.set_migration_rate`
+method.
+This allows us to specify the expected number of migrants moving
+from population `dest` to population `source` per generation, divided by the size of
+population `source`.  When this rate is small (close to 0), it is approximately
+equal to the fraction of population `source` that consists of new migrants from
+population `dest` in each generation.
+
+```{note}
+The reason for this (perhaps) counter-intuitive specification of `source` and `dest` is that ``msprime`` simulates backwards-in-time. See [this](https://tskit.dev/msprime/docs/latest/demography.html#direction-of-time) for further explanation.
+```
 
 For instance, the following migration matrix specifies that in each generation,
 approximately 5% of population 0 consists of migrants from population 1, and
 approximately 2% of population 1 consists of migrants from population 0.
 
 
 ```{code-cell} ipython3
-M = np.array([
-[0, 0.05],
-[0.02, 0]])
-dem = msprime.Demography(populations=[pop0, pop1], migration_matrix=M)
-ts = msprime.sim_ancestry(samples=[samples0, samples1], demography=dem, sequence_length=1000, random_seed=17, recombination_rate=1e-7)
+dem = msprime.Demography()
+dem.add_population(
+  name="Population0",
+  description="Plotted in red.",
+  initial_size=500,
+  growth_rate=0
+  )
+dem.add_population(
+  name="Population1",
+  description="Plotted in blue.",
+  initial_size=500,
+  growth_rate=0
+  )
+dem.set_migration_rate(source=0, dest=1, rate=0.05)
```

Comment:
So we're trying to encourage people to **name** their populations and use these names in the specification rather than having to remember what 0 and 1 refer to. How about we change the names here to "A" and "B" (or something) and then we can say
```
dem.set_migration_rate(source="A", dest="B", rate=0.05)
```

Reply:
The names "Population0" is unwieldy, and doesn't encourage use. It's not any more informative really than "A" I think.

Reply:
The reason why I didn't do this is because in the tikz plots here, everything is labelled "Population0" and "Population1". I'll have a look at changing these later.

Reply:
Ah, ok.